aspnetcore/release-notes/aspnetcore-5.0.md

```diff
@@ -0,0 +1,476 @@
+
+## gRPC
+
+Many preformance improvements have been made in [gRPC](https://grpc.io/). <!-- For more information, see [this ]
+Waiting for https://github.com/aspnet/specs/pull/523
+--> 
+For more information, see <xref:grpc/index>.
+
+## SignalR
+
+SignalR Hub filters, called Hub pipelines in ASP.NET SignalR, is a feature that allows code to run before and after Hub methods are called. Running code before and after Hub methods are called is similar to how middleware has the ability to run code before and after an HTTP request. Common uses include logging, error handling, and argument validation.
+
+For more information, see [Use hub filters in ASP.NET Core SignalR](xref:signalr/hub-filters).
+
+### SignalR parallel hub invocations
+
+ASP.NET Core SignalR is now capable of handling parallel hub invocations. The default behavior can be changed to allow clients to invoke more than one hub method at a time:
+
+[!code-csharp[](~/release-notes/sample/StartupSignalRhubs.cs?name=snippet)]
+
+### Added Messagepack support in SignalR Java client
+
+A new package, [com.microsoft.signalr.messagepack](https://search.maven.org/search?q=g:com.microsoft.signalr.messagepack), adds MessagePack support to the SignalR Java client. To use the MessagePack hub protocol, add `.withHubProtocol(new MessagePackHubProtocol())` to the connection builder:
+
+```java
+HubConnection hubConnection = HubConnectionBuilder.create(
+                           "http://localhost:53353/MyHub")
+               .withHubProtocol(new MessagePackHubProtocol())
+               .build();
+```
+
+<!--
+See [Update SignalR code](xref:migration/31-to-50#signalr) for migration instructions.
+-->
+
+## Kestrel
+
+* Reloadable endpoints via configuration: Kestrel can detect changes to configuration passed to [KestrelServerOptions.Configure](xref:Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerOptions.Configure%2A) and unbind from existing endpoints and bind to new endpoints without requiring an application restart.
+* HTTP/2 response headers improvements. For more information, see [Performance improvements](#performance-improvements) in the next section.
+* Support for additional endpoints types in the sockets transport: Adding to the new API introduced in <xref:System.Net.Sockets?displayProperty=nameWithType>, the sockets default transport in [Kestrel](xref:fundamentals/servers/kestrel) allows binding to both existing file handles and Unix domain sockets. Support for binding to existing file handles enables using the existing `Systemd` integration without requiring the `libuv` transport.
+* Custom header decoding in Kestrel: Apps can specify which <xref:System.Text.Encoding> to use to interpret incoming headers based on the header name instead of defaulting to `UTF-8`. Set the <xref:System.Net.Http.SocketsHttpHandler.RequestHeaderEncodingSelector> property on <xref:Microsoft.AspNetCore.Server.Kestrel.KestrelServerOptions> to specify which encoding to use:
```

Comment:
Kestrel's selector is a `Func<string, Encoding>` while SocketsHttpHandler's selector is a `HeaderEncodingSelector<System.Net.Http.HttpRequestMessage>`.

```suggestion
* Custom header decoding in Kestrel: Apps can specify which <xref:System.Text.Encoding> to use to interpret incoming headers based on the header name instead of defaulting to `UTF-8`. Set the <xref:Microsoft.AspNetCore.Server.Kestrel.KestrelServerOptions.RequestHeaderEncodingSelector> property to specify which encoding to use:
```